aml/src/term_object.rs

```diff
@@ -877,6 +878,98 @@ where
         .map(|((), result)| Ok(result))
 }
 
+fn index_field_element<'a, 'c>(
+    index_handle: AmlHandle,
+    data_handle: AmlHandle,
+    flags: FieldFlags,
+    current_offset: u64,
+) -> impl Parser<'a, 'c, u64>
+where
+    'c: 'a,
+{
+    /*
+     * Reserved fields shouldn't actually be added to the namespace; they seem to show gaps in
+     * the operation region that aren't used for anything.
+     *
+     * NOTE: had to use raw_pkg_length() instead of pkg_length() here because pkg_length() performs
```

Comment:
Yeah, I've created a separate issue for that already. The parsing of the field might become a little more complicated, as instead of just pkg_length() we would need to also pass a opreg to the parser

Reply:
As for implemeting as_* functions on IndexFields, I've done that on my separate branch where I just try to make a dirty but fully working implementation for my laptop, but that involves making almost everything (write_field/write_region/etc.) take an immutable ref to an AmlContext instead of a mutable one which doesn't seem quite right.

I've also tried making as_integer take a mutable ref, but then we'd have a problem in some places where pieces of code like:

```
// in def_increment()/def_decrement() parsers
let value = try_with_context!(context, context.read_target(&addend));
let value = try_with_context!(context, value.as_integer(context));
```

fails to compile because the first line now borrows context immutably and the second one tries to borrow it mutably. Fixing that in and preserving mutability of the ref might involve either moving the context inside some kind of Spinlock or making wrapper functions that merge the functionality of read_target() and as_* without needing to borrow twice.